### PR TITLE
EasyConfig for installing Molcas from precompiled binaries

### PR DESCRIPTION
**Pull request by valtandor**
Note that the BinariesTarball Easyblock must be upgraded before testing this EasyConfig.

### CODE REVIEW

**Comment by valtandor** [comment]
Not yet.

**Comment by valtandor** [comment]
Actually, it seems further changes for this may be required. The precompiled Molcas binary seems to have some strange assumptions about where files are to be.

**Comment by boegel** [comment]
@valtandor: I've added a `(WIP)` tag in the PR title, do remove (and drop in a comment) when you feel this is good to go

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by valtandor** [comment]
My understanding is that this Easyconfig now works. I've used a different EasyBlock and rewritten the config to do what needs to be done.

File: easybuild/easyconfigs/m/Molcas/Molcas-8.0-15.06.18_CentOS_6.6_x86_64.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+description = """Molcas is an ab initio quantum chemistry software package
+developed by scientists to be used by scientists. The basic philosophy is is to
+be able to treat general electronic structures for molecules consisting of
+atoms from most of the periodic table. As such, the primary focus of the
+package is on multiconfigurational methods with applications typically
+connected to the treatment of highly degenerate states."""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+source_urls = ['http://www.kvant.kemi.uu.se/molcas/']
```
Comment (boegel) [inline easybuild/easyconfigs/m/Molcas/Molcas-8.0-15.06.18_CentOS_6.6_x86_64.eb]: downloading requires registration, so there's little point in including this?

including it as a comment makes sense though

File: easybuild/easyconfigs/m/Molcas/Molcas-8.0-15.06.18_CentOS_6.6_x86_64.eb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+sources = ['%(namelower)s-%(version)s%(versionsuffix)s.tgz']
+unpack_options = "--transform='s,^%(namelower)s-%(version)s%(versionsuffix)s,%(namelower)s_files,'"
```
Comment (boegel) [inline easybuild/easyconfigs/m/Molcas/Molcas-8.0-15.06.18_CentOS_6.6_x86_64.eb]: what's this about? please add a comment in the easyconfig to explain

File: easybuild/easyconfigs/m/Molcas/Molcas-8.0-15.06.18_CentOS_6.6_x86_64.eb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    'mkdir -p %(installdir)s/bin',
+    'cp %(installdir)s/%(namelower)s_files/sbin/molcas.driver %(installdir)s/bin/molcas',
+]
+
+sanity_check_paths = {
+    'files': ['bin/molcas'],
+    'dirs': ['bin'],
+}
+
+moduleclass = 'chem'
```
Comment (boegel) [inline easybuild/easyconfigs/m/Molcas/Molcas-8.0-15.06.18_CentOS_6.6_x86_64.eb]: make this the last line please, to be consistent with other easyconfigs

File: easybuild/easyconfigs/m/Molcas/Molcas-8.0-15.06.18_CentOS_6.6_x86_64.eb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+modextravars = {'MOLCAS': '%(installdir)s/%(namelower)s_files'}
```
Comment (boegel) [inline easybuild/easyconfigs/m/Molcas/Molcas-8.0-15.06.18_CentOS_6.6_x86_64.eb]: should be separated with an empty line from the `moduleclass` line

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 0 out of 0 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/065d69ef2690de770ae1 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/42ce97fea0b2767d7efc for a full test report.

**Comment by boegel** [comment]
@valtandor: works like a charm, but let's fix the small style remarks before merging this

File: easybuild/easyconfigs/m/Molcas/Molcas-8.0-15.06.18_CentOS_6.6_x86_64.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/m/Molcas/Molcas-8.0-15.06.18_CentOS_6.6_x86_64.eb]: @valtandor: https://github.com/valtandor/easybuild-easyconfigs/pull/4 fixes the remarks, except for this one

I'm not sure why this needs to be included, can you add a comment on the line above this one to explain why this is needed?

**Comment by valtandor** [comment]
@boegel: I trust the comments I put in are satisfactory.

**Comment by JensTimmerman** [comment]
@boegel anything holding this back?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/78e0d1cf62942f06495f073699fcd01a for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @valtandor, sorry for the long wait, not sure why it went under the radar...

**Pull request merged**